src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -574,6 +576,14 @@ FixedwingPositionControl::landing_status_publish()
 	_pos_ctrl_landing_status_pub.publish(pos_ctrl_landing_status);
 }
 
+float FixedwingPositionControl::getCorrectedNpfgRollSetpoint()
+{
+	// Scale the npfg output if npfg is not certaiin for correct output
+	float new_roll_setpoint(_npfg.getRollSetpoint());
+	const float can_run_factor(_npfg.canRun(_local_pos, _wind_valid));
+	return _last_roll_setpoint + can_run_factor * (new_roll_setpoint - _last_roll_setpoint);
```

Comment:
are you sure we want to hold the last setpoint and not just regulate to zero? this could result in some spinning maneuvers, where as a zero roll setpoint at least would just level out and stabilize until it has reasonable inputs to work with.

Reply:
Yes, regulate to zero is potentially more secure, i will change it. Only downside of this would be that you could turn against the wind and then stay there since you level out. And when you don't change anymore, then you probably will never get a wind estimate. But that should be handled elsewhere.